luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/DefaultMetricFactory.java

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.config.luthier.factories;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.yahoo.bard.webservice.config.luthier.Factory;
+import com.yahoo.bard.webservice.config.luthier.LuthierIndustrialPark;
+import com.yahoo.bard.webservice.config.luthier.LuthierValidationUtils;
+import com.yahoo.bard.webservice.data.config.metric.makers.MetricMaker;
+import com.yahoo.bard.webservice.data.metric.LogicalMetric;
+import com.yahoo.bard.webservice.data.metric.LogicalMetricInfo;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A Factory that supports all aggregation metrics, including but not limited to: longSum, doubleMax, longMin, etc.
+ */
+public class DefaultMetricFactory implements Factory<LogicalMetric> {
+    private static final String ENTITY_TYPE = "AggregationMetric";
+    private static final String MAKER = "maker";
+    private static final String LONG_NAME = "longName";
+    private static final String CATEGORY = "category";
+    private static final String DESCRIPTION = "description";
+    private static final String DATA_TYPE = "dataType";
+    private static final String DEPENDENCY_METRIC_NAMES = "dependencyMetricNames";
+
+    @Override
+    public LogicalMetric build(String name, ObjectNode configTable, LuthierIndustrialPark resourceFactories) {
+        validateFields(
+                name,
+                configTable,
+                Arrays.asList(MAKER, LONG_NAME, CATEGORY, DESCRIPTION, DEPENDENCY_METRIC_NAMES)
+        );
+        MetricMaker maker = resourceFactories.getMetricMaker(configTable.get(MAKER).textValue());
```

Comment:
One thing that we're missing in the system is some null checking around dependencies. For example,
what happens if someone types `longSumm` (note the extra `m`) instead of `longSum`? We should have
an error message "Unknown maker `longSumm`, avaiable makers are `list-of-maker-names`. But I'm not sure
we do. I think we might just null pointer somewhere in our system. We should think about tightening
that up after this PR. One option is that the Factory is expected to throw an exception if it can't find
the configuration for the given name, and the LuthierIndustrialPark catches the exception and prints an
error. That way, we have this family of errors handled in the same place, instead of scattered about
all the factories.

Reply:
I can somehow envision that, but I will hold it off for now as I believe it might deserve a more thorough view when we get back together. And it is certainly not a feature we absolutely have to support in the first release. 
I'd still love to get _some_ of it done before the Summer ends, though.